File: scripts/algorithms/S/Sum of Beauty of All Substrings/Sum of Beauty of All Substrings.py
class Solution:
    def beautySum(self, s: str) -> int:
        c, n, ans = Counter(s), len(s), 0
        for i in range(n-2):
            x=c.copy()
            for j in range(n-1,i+1,-1):
                ans+=max(x.values())-min(x.values())
                if x[s[j]]==1:
                    del x[s[j]]
                else:
                    x[s[j]]-=1
            if c[s[i]]==1:
                del c[s[i]]
            else:
                c[s[i]]-=1
        return ans

File: scripts/algorithms/S/Sum of Beauty of All Substrings/Sum of Beauty of All Substrings.js
// Runtime: 667 ms (Top 81.08%) | Memory: 48.8 MB (Top 62.16%)
var beautySum = function(s) {
    const len = s.length;

    let ans = 0;
    const freq = new Array(26).fill(0);

    for(let i = 0; i < len; i++) {
        freq[s[i].charCodeAt(0) - 'a'.charCodeAt(0)]++;
        for(let j = i + 1; j < len; j++) {
            freq[s[j].charCodeAt(0) - 'a'.charCodeAt(0)]++;
            ans += minMaxDiff(freq);
        }
        freq.fill(0);
    }

    function minMaxDiff(freq) {
        freq = freq.filter(f => f != 0);
        return Math.max(...freq) - Math.min(...freq);
    }

    return ans;
};